### ISSUE TITLE
VS Live Share integration in Azure Data Studio

### ISSUE THREAD

**Issue by samirbehara-zz** [microsoft/azuredatastudio#869]
- SQL Operations Studio Version: Future

Visual Studio Live Share is an awesome collaboration tool for developers and is currently in private preview. Since SQL Operations Studio is built on top of VSCode, it would be really nice to have this tool ported to SQL Ops Studio in future, once VS Live Share is ready.

This is just a suggestion & I wanted to get this into GitHub so that it can be planned/discussed or thought about.

**Comment by jzabroski** [comment]
It looks like VSCode already has a Live Share extension in the VS Gallery. And that Live Share is on GitHub: https://github.com/MicrosoftDocs/live-share

**Comment by hinchley2018** [comment]
New to this repo. What work is involved in supporting this extension in Azure Data Studio?
Maybe all of us interested in this feature could collaborate?
It would definitely solve some problems for me in my projects. But I don't know where you would even start.

**Comment by hinchley2018** [comment]
@yualan or @jzabroski any tips on how i could begin working on integrating live share into ADS? My team could really use this feature while everyone is working remotely.

**Comment by jzabroski** [comment]
@hinchley2018 I don't work for MSFT but:

A workaround that might be suitable to you is to use [vscode](https://github.com/Microsoft/vscode) + [vscode-mssql](https://github.com/Microsoft/vscode-mssql) extension:

<img src="https://github.com/Microsoft/vscode-mssql/raw/master/images/mssql-demo.gif" alt="demo" style="width:480px;"/>

**Comment by hinchley2018** [comment]
This might work in the meantime. 

I would still like to know how I can contribute to a feature like this live share integration. Perhaps I'll poke around VSCode's & live share codebases to try to understand how that existing integration works.

**Comment by danielsbezerra** [comment]
Any chance we can have it any time soon? It would be so helpful for my DBA team.

**Comment by henry-js** [comment]
You can download the LiveShare extensions from [here](https://marketplace.visualstudio.com/items?itemName=MS-vsliveshare.vsliveshare) and [here](https://marketplace.visualstudio.com/items?itemName=MS-vsliveshare.vsliveshare-audio).
 I downloaded and installed both to Azure Data Studio fine:
![image](https://user-images.githubusercontent.com/79054685/179977740-8d6869b7-39ae-41eb-baf1-1a8f944e02f6.png)

**Comment by danielsbezerra** [comment]
> > It looks like Microsoft might be closing off these types of features to unpaid customers. [OmniSharp/omnisharp-vscode#5276](https://github.com/OmniSharp/omnisharp-vscode/issues/5276)
> 
> You can download the LiveShare extensions from [here](https://marketplace.visualstudio.com/items?itemName=MS-vsliveshare.vsliveshare) and [here](https://marketplace.visualstudio.com/items?itemName=MS-vsliveshare.vsliveshare-audio). I downloaded and installed both to Azure Data Studio fine: ![image](https://user-images.githubusercontent.com/79054685/179977740-8d6869b7-39ae-41eb-baf1-1a8f944e02f6.png)

Works perfectly! In fact, there's an option to download Live Share Extension Pack from [here](https://marketplace.visualstudio.com/items?itemName=MS-vsliveshare.vsliveshare-pack), which includes live share audio and text chat.

<sup>Note.: Instructions about how to download extension directly from Marketplace [here](https://code.visualstudio.com/docs/editor/extension-marketplace#_can-i-download-an-extension-directly-from-the-marketplace).</sup>

**Comment by magnusbratt** [comment]
How do you install the extension to ADS? When I use the links provided above it opens in VS Code.

**Comment by henry-js** [comment]
> How do you install the extension to ADS? When I use the links provided above it opens in VS Code.

@magnusbratt the previous commenter provided a note on downloading/installing extension files directly from the marketplace. The instructions are the same for ADS:

>## Can I download an extension directly from the Marketplace?
>Some users prefer to download an extension once from the Marketplace and then install it multiple times from a local share. >This is useful when there are connectivity concerns or if your development team wants to use a fixed set of extensions.
>
>To download an extension, navigate to the details page for the specific extension within the [Marketplace](https://marketplace.visualstudio.com/vscode). On that page, there is a Download Extension link in the Resources section, which is located on the right-hand side of the page.
>
>Once downloaded, you can then install the extension via the Install from VSIX command in the Extensions view command dropdown.

![image](https://user-images.githubusercontent.com/79054685/192788932-e8f94a96-df37-461d-a637-3f5f2a48a6bd.png)

**Comment by mbellas-dumac** [comment]
I've got both extensions installed but can't get it to sign in, anyone have a solution? I always get `Auth token returned by microsoft provider is empty`

**Comment by alanrenmsft** [comment]
Thank you for your feedback! This feature is currently out of scope and we do not plan to work on it in a currently planned release. We will close this issue to keep our backlog focused on requests that we are planning to work on. Please note that users can continue to vote and comment on closed issues, which we encourage as it helps us understand user interest and can provide more details about why a feature is requested.

**Comment by mikhail-ict** [comment]
@henry-js Is it still working for you? There seems to be an authorisation issue with the current ADS and Live Share versions...
